portfolio/src/main/webapp/script.js

```diff
@@ -26,3 +26,98 @@ function addRandomGreeting() {
   const greetingContainer = document.getElementById('greeting-container');
   greetingContainer.innerText = greeting;
 }
+
+function showPicture(imgs) {
+  var expandedImg = document.getElementById("expandedImg");
+  var imgText = document.getElementById("imgtext");
+  expandedImg.src = imgs.src;
+  imgText.innerHTML = imgs.alt;
+  expandImg.parentElement.style.display = "block";
+}
+
+
+
+
+
+
+
+/**
+ * Fetches a random quote from the server and adds it to the DOM.
+ */
+function sayHello() {
+  console.log('Fetching hello.');
+
+  // The fetch() function returns a Promise because the request is asynchronous.
+  const responsePromise = fetch('/data');
+
+  // When the request is complete, pass the response into handleResponse().
+  responsePromise.then(handleResponse);
+}
+
+/**
+ * Handles response by converting it to text and passing the result to
+ * addQuoteToDom().
+ */
+function handleResponse(response) {
+  console.log('Handling the response.');
+
+  // response.text() returns a Promise, because the response is a stream of
+  // content and not a simple variable.
+  const textPromise = response.text();
+
+  // When the response is converted to text, pass the result into the
+  // addQuoteToDom() function.
+  textPromise.then(replyHello);
+}
+/** Adds a random quote to the DOM. */
+function replyHello(helloStr) {
+  console.log('Hello Darya');
+
+  const quoteContainer = document.getElementById('hello-sell');
+  quoteContainer.innerText = helloStr;
+}
+
+/**
+ * The above code is organized to show each individual step, but we can use an
+ * ES6 feature called arrow functions to shorten the code. This function
+ * combines all of the above code into a single Promise chain. You can use
+ * whichever syntax makes the most sense to you.
+ */
+function sayHelloUsingArrowFunctions() {
+  fetch('/data').then(response => response.text()).then((helloStr) => {
+    document.getElementById('hello-sell').innerText = helloStr;
+  });
+}
+
+/**
+ * Another way to use fetch is by using the async and await keywords. This
+ * allows you to use the return values directly instead of going through
+ * Promises.
+ */
+async function sayHelloUsingAsyncAwait() {
+  const response = await fetch('/data');
+  const quote = await response.text();
+  document.getElementById('hello-sell').innerText = quote;
+}
+
+function getServerStats() {
+  fetch('/data').then(response => response.json()).then((jsonStr) => {
+    // stats is an object, not a string, so we have to
+    // reference its fields to create HTML content
+
+    const statsListElement = document.getElementById('hello-sell');
+    statsListElement.innerHTML = '';
+    statsListElement.appendChild(
+        createListElement('First string: ' + jsonStr[0]));
+    statsListElement.appendChild(
+        createListElement('Second string: ' + jsonStr[1]));
+    statsListElement.appendChild(
+        createListElement('Third string: ' + jsonStr[2]));
+  });
+}
```

Comment:
You could use `async` and `await` here instead of promise.then():

```javascript
  function getServerStats() {
    const response = await fetch('/data');
    const jsonStr = await response.json();
    ...
  }
```